Scripts/ImageFactory/Scripts/MakeVM.ps1

```diff
@@ -25,9 +22,53 @@
     [System.Security.SecureString] $machinePassword,
 
     [Parameter(Mandatory=$true, HelpMessage="The name of the lab")]
-    [string] $vmSize
+    [string] $vmSize,
+
+    [boolean] $includeSysprep = $false
 )
 
+function makeUpdatedTemplateFile ($origTemplateFile, $outputFile)
+{
+    $armTemplate= ConvertFrom-Json -InputObject (gc $origTemplateFile -Raw -Encoding Ascii)
+
+    #add the Sysprep or deprovision artifact to the list of artifacts for the VM
+    $newArtifact = @{}
+    if ($armTemplate.resources[0].properties.galleryImageReference.osType -eq 'Windows')

```

Comment:
$armTemplate.resources[0].properties.galleryImageReference.osType is not guaranteed to be in the armTemplate.

Many of my templates are built from custom images, not from gallery references.  

How else can we do this?